### DISCUSSION TITLE
IPsec session keys

### DISCUSSION THREAD

**Discussion by hnsantu554** [strongswan/strongswan#2478] [Q&A]
Ip xfrm state in linux shows critical information like AH/ESP keys which is used for encryption and decryption, this can be security concern.
How do we protect key information?
I am using linux kernel 5.4.
Please help , thank you

**Comment by tobiasbrunner** [comment]
While only privileged users/processes can query the SAs, it could be a valid concern. So with newer kernels the keys can be hidden from userland (XFRM lockdown feature, `/sys/kernel/security/lockdown`). This requires at least a 5.11 kernel and it does have some implications on strongSwan as dynamically updating CHILD_SAs via MOBIKE won't work anymore and there might be some extra log messages (see #2156).

**Reply by hnsantu554** [reply]
Hi Tobiasbrunner,
Currently we are using linux 5.4 where kernel lockdown is not supported ,since ip xfrm state provides key information , can we restrict the ip xfrm state command only to a specific users to execute , will it affect strongswan operation? It will be really helpful if you provide some points on this.
Thanks

**Reply by tobiasbrunner** [reply]
> can we restrict the ip xfrm state command only to a specific users to execute 

Only the root user (or a process with NET_ADMIN capability, like strongSwan requires) is allowed to access the XFRM interface (i.e. a regular user can't execute the command).
